custom_components/tahoma/climate_st.py

```diff
@@ -0,0 +1,242 @@
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return MAP_HVAC_MODES[self.select_state(ST_DEROGATION_TYPE_STATE)]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return MAP_PRESET_MODES[
+            self.select_state(ST_HEATING_MODE_STATE)
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
+        ]
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return [
+            PRESET_NONE,
+            PRESET_FREEZE,
+            PRESET_NIGHT,
+            PRESET_AWAY,
+            PRESET_HOME,
+        ]
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 15.0
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 26.0
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return (
+            self.select_state(PRESET_TEMPERATURES[self.preset_mode])
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(CORE_DEROGATED_TARGET_TEMPERATURE_STATE)
+        )
+
+    def set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self.apply_action(
+            COMMAND_SET_DEROGATION, temperature, STATE_DEROGATION_FURTHER_NOTICE
+        )
+        self.apply_action(
+            COMMAND_SET_MODE_TEMPERATURE, STATE_PRESET_MANUAL, temperature
+        )
+        self.apply_action(COMMAND_REFRESH_STATE)
```

Comment:
Why refresh the state for this platform in particular?

Reply:
```
COMMAND_REFRESH_DEROGATION = "refreshDerogation"
COMMAND_REFRESH_MODE_SETTING = "refreshModeSettings"
COMMAND_REFRESH_STATE = "refreshState"
```
these commands are part of ST but not AEH. To be fair, I'm not sure what it does, but at some point, TaHoma wasn't updating the states on its side without triggering the refreshState.
I'll try without it and see whether it should be kept.

Reply:
After some testing, the `refreshState` is needed to have the new state reflected in TaHoma, thus in HA.
plus opening the thermostat popup in tahomalink and sending a command triggers the refresh as well.
![image](https://user-images.githubusercontent.com/5701372/88025162-26ec9c00-cb34-11ea-9e11-bf1a9cdfbdee.png)

Reply:
Understood. This refresh command is not sent for the other widgets?

It would be nice to group our commands in one call like Somfy does.

Reply:
I think it's specific to the somfy smart thermostat.